pkg/service/credential/storage.go

```diff
@@ -504,19 +504,20 @@ func (cs *Storage) GetStatusListCredentialsByIssuerSchemaPurpose(ctx context.Con
 	}
 
 	// now get each credential by key
-	var storedCreds []StoredCredential
+	storedCreds := make([]StoredCredential, 0, len(issuerSchemaKeys))
 	for _, key := range issuerSchemaKeys {
 		credBytes, err := cs.db.Read(ctx, statusListCredentialNamespace, key)
 		if err != nil {
 			logrus.WithError(err).Errorf("could not read credential with key: %s", key)
-		} else {
-			var cred StoredCredential
-			if err = json.Unmarshal(credBytes, &cred); err != nil {
-				logrus.WithError(err).Errorf("unmarshalling credential with key: %s", key)
-			}
+			return nil, err
```

Comment:
This wasn't erroring out. Which ended up causing some bizarre failures in the e2e tests.

Reply:
good find